### PR TITLE
tests(ci): copy and upload failed test logs

### PR DESCRIPTION
**Pull request by Water-Melon**
KAG-2563

### Checklist

- [ ] The Pull Request has tests
- [ ] A changelog file has been created under `changelog/unreleased/kong` or `skip-changelog` label added on PR if changelog is unnecessary. [README.md](https://github.com/Kong/gateway-changelog/blob/main/README.md)
- [ ] There is a user-facing docs PR against https://github.com/Kong/docs.konghq.com - PUT DOCS PR HERE

### Issue reference

Fix #_[issue number]_

### CODE REVIEW

**Comment by Water-Melon** [comment]
https://github.com/Kong/kong/actions/runs/9326875800/job/25676004254?pr=13115
errlogs uploaded

**Comment by Water-Melon** [comment]
https://github.com/Kong/kong/actions/runs/9343590417/job/25713428497?pr=13115#step:21:1
errlogs uploaded

**Review by ADD-SP — APPROVED** [review]

**Comment by ADD-SP** [comment]
Could you resolve merge conflicts?

**Comment by Water-Melon** [comment]
> Could you resolve merge conflicts?

It's been fixed.

**Comment by Water-Melon** [comment]
> I downloaded the file from https://github.com/Kong/kong/actions/runs/9343590417?pr=13115, the filename is a bit hard to understand:
> 
> `...zel-bin_build_kong-dev_share_lua_5.1_luassert_assert.lua.43.log`
> 
> Can we consider change the filename to something like:
> 
> ```
> spec/02-integration/06-invalidations/02-core_entities_invalidations_spec_line_109.log
> ```
> 
> or
> 
> ```
> spec_02-integration_06-invalidations_02-core_entities_invalidations_spec_line_109.log
> ```
> 
> instead? It will be much easier to search for.

That was because I encapsulated the logic for copying errlog into a function before opening the PR, but the level for `debug.getinfo` was not updated accordingly. This problem has been fixed, and I also adjusted the name suffix.

Here is the test:
https://github.com/Kong/kong/actions/runs/9411253941/job/25924210228?pr=13115#step:20:42

File: spec/helpers.lua [hunk 1]
```diff
@@ -2473,6 +2474,31 @@ local deep_sort do
...
+
+  if string.find(file_path, '/', nil, true) then
+    file_path = string.gsub(file_path, '/', '_')
+  end
+  file_path = errlog_cache_dir .. "/" .. file_path:gsub("%.lua$", "_") .. "line_" .. line_number .. '.log'
+
+  ok = pl_file.copy(errlog_path, file_path)
+  if not ok then
+    print(colors("%{yellow}Log saved as: " .. file_path .. "%{reset}"))
+  end
```
Comment (fffonion) [inline spec/helpers.lua:2499]:
```suggestion
  ok, err = pl_file.copy(errlog_path, file_path)
  if ok then
    print(colors("%{yellow}Log saved as: " .. file_path .. "%{reset}"))
  else
    print(colors("%{red}Failed to save error log for test " .. file_path .. ": " .. err))
  end
```

Comment (Water-Melon) [reply spec/helpers.lua]: It's been fixed. Thanks very much!

**Review by fffonion — APPROVED** [review]

**Comment by Water-Melon** [comment]
Here is the newest test ci: https://github.com/Kong/kong/actions/runs/9462602728/job/26065965800?pr=13115

We can see that the first test outputs a red 'Failed to save...' message because there is no error log. The second test outputs a yellow 'Log saved as...' message because an error log is present.

**Pull request merged**